### PR TITLE
change LazySet to ConvexSet

### PR DESCRIPTION
**Pull request by lucaferranti**
x-ref: https://github.com/JuliaReach/LazySets.jl/issues/1895#issuecomment-1171612637

- [x] change `LazySet` --> `ConvexSet`
- [x] add supertype `LazySet`
- [x] ~~make nonconvex sets subtypes of `LazySet`~~ for upcoming PR
      - [ ] DensePolynomialZonotope
      - [ ] SimpleSparsePolynomialZonotope
      - [ ] UnionSet
      - [ ] UnionSetArray
      - [ ] Complement
      - [ ] Rectification

### CODE REVIEW

**Comment by schillic** [comment]
The file `LazySet.jl` needs to be renamed. And there should be a new supertype `LazySet`.

I changed this PR to a draft to indicate that we should not yet merge it.

**Comment by mforets** [comment]
> make nonconvex sets subtypes of LazySet

I think this change will make the PR unnecessarily big, let's add defer to another PR?

File: src/Interfaces/LazySet.jl [hunk 1]
```diff
@@ -1,55 +1,16 @@
...
-elements ``x, y ∈ S`` and ``0 ≤ λ ≤ 1`` it holds that ``λ·x + (1-λ)·y ∈ S``.
+Abstract type for the set types in LazySets.jl
 
 ### Notes
 
 `LazySet` types should be parameterized with a type `N`, typically `N<:Real`,
 for using different numeric types.
 
-Every concrete `LazySet` must define the following functions:
+Every concrete `ConvexSet` must define the following functions:
```
Comment (schillic) [inline src/Interfaces/LazySet.jl:13]:
```suggestion
Every concrete `LazySet` must define the following functions:
```

File: src/Interfaces/LazySet.jl [hunk 2]
```diff
@@ -60,1559 +21,5 @@ The function
 is optional because there is a fallback implementation relying on `σ`.
 However, for unbounded sets (which includes most lazy set types) this fallback
 cannot be used and an explicit method must be implemented.
-
-The subtypes of `LazySet` (including abstract interfaces):
-
-```jldoctest; setup = :(using LazySets: subtypes)
-julia> subtypes(LazySet, false)
-14-element Vector{Any}:
- AbstractAffineMap
```
Comment (schillic) [inline src/Interfaces/LazySet.jl:69]: I think this docstring should stay with `LazySet`.

Comment (lucaferranti) [reply src/Interfaces/LazySet.jl]: I'm not sure about that, `subtypes(LazySet, true)` seems to be recursive and would also print all subtypes of `ConvexSet`, which are already listed in `ConvexSet` docstring, this way it synthetically lists `LazySet` direct subtypes, `ConvexSet` and (one day) the non-convex sets in the package.

Comment (schillic) [reply src/Interfaces/LazySet.jl]: Yes, remove them from `ConvexSet` of course.

**Comment by schillic** [comment]
LGTM, thanks! I will approve next week after doing some other refactoring. We should also merge the other recently opened PRs before merging this PR, but I think they do not conflict.

**Review by mforets — APPROVED** [review]

**Review by schillic — APPROVED** [review]

**Pull request merged**